lib/Runtime/Types/TypePath.h

```diff
@@ -61,45 +52,71 @@ namespace Js
     class TypePath
     {
         friend class PathTypeHandlerBase;
-        friend class DynamicObject;
-        friend class SimplePathTypeHandler;
-        friend class PathTypeHandler;
-
     public:
-        static const uint MaxPathTypeHandlerLength = MAX_SIZE_PATH_LENGTH;
-        static const uint InitialTypePathSize = 16;
+        // This is the space between the end of the TypePath and the allocation granularity that can be used for assignments too.
+#ifdef SUPPORT_FIXED_FIELDS_ON_PATH_TYPES
+#if defined(_M_X64_OR_ARM64)
+#define TYPE_PATH_ALLOC_GRANULARITY_GAP 0
+#else
+#define TYPE_PATH_ALLOC_GRANULARITY_GAP 2
+#endif
+#else
+#if defined(_M_X64_OR_ARM64)
+#define TYPE_PATH_ALLOC_GRANULARITY_GAP 1
+#else
+#define TYPE_PATH_ALLOC_GRANULARITY_GAP 3
```

Comment:
Unfortunately no. The constant is needed before the class TypePath is fully defined.
I already added compiler assert at the end to validate this constant.